evap/evaluation/tools.py

```diff
@@ -58,52 +60,52 @@ def is_prefetched(instance, attribute_name: str):
     return False
 
 
-def discard_cached_related_objects(instance):
+def discard_cached_related_objects(instance: M) -> M:
     """
     Discard all cached related objects (for ForeignKey and M2M Fields). Useful
     if there were changes, but django's caching would still give us the old
     values. Also useful for pickling objects without pickling the whole model
     hierarchy (e.g. for storing instances in a cache)
     """
     # Extracted from django's refresh_from_db, which sadly doesn't offer this part alone (without hitting the DB).
-    for field in instance._meta.concrete_fields:
+    for field in instance._meta.concrete_fields:  # type: ignore
         if field.is_relation and field.is_cached(instance):
             field.delete_cached_value(instance)
 
-    for field in instance._meta.related_objects:
+    for field in instance._meta.related_objects:  # type: ignore
         if field.is_cached(instance):
             field.delete_cached_value(instance)
 
-    instance._prefetched_objects_cache = {}
+    instance._prefetched_objects_cache = {}  # type: ignore
 
     return instance
 
 
-def is_external_email(email):
+def is_external_email(email: str) -> bool:
     return not any(email.endswith("@" + domain) for domain in settings.INSTITUTION_EMAIL_DOMAINS)
 
 
-def sort_formset(request, formset):
+def sort_formset(request: HttpRequest, formset: BaseFormSet) -> None:
     if request.POST:  # if not, there will be no cleaned_data and the models should already be sorted anyways
         formset.is_valid()  # make sure all forms have cleaned_data
```

Comment:
(not yours), but?
```suggestion
        assert formset.is_valid()  # make sure all forms have cleaned_data
```

Reply:
One test fails then. The method seems to be intended to also be callable with invalid formsets (via the default value in the line below).